### PR TITLE
Update doc for 3.0.0 release

### PR DESCRIPTION
**Pull request by kevinlind**
## Description

Update the GitHub docs for 3.0.0 release.

- Removes `registerExtension` from api-reference as the API is removed in 3.0.0.
- Updates getting-started to use `3.+` in dependency listing.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: Documentation/getting-started.md [hunk 1]
```diff
@@ -45,9 +45,9 @@ The Edge Network extension depends on the following extensions:
 1. Add the Mobile Core, Identity for Edge Network, and Edge Network extensions to your project using the app's Gradle file:
 
 ```gradle
-implementation 'com.adobe.marketing.mobile:core:2.+'
-implementation 'com.adobe.marketing.mobile:edge:2.+'
-implementation 'com.adobe.marketing.mobile:edgeidentity:2.+'
+implementation 'com.adobe.marketing.mobile:core:3.+'
```
Comment (emdobrin) [inline Documentation/getting-started.md:48]: let's use bom samples instead

Comment (cacheung) [reply Documentation/getting-started.md:48]: bom sample won't work until we release.  Should we keep it as 3.+ for now and update it along with build.gradle in sample apps all together before release?

Comment (kevinlind) [reply Documentation/getting-started.md:48]: @cacheung We can update it now as it's our documentation and not the build config for an application. This doc won't go "live" until we release so it's okay to update it now.

File: Documentation/getting-started.md [hunk 2]
```diff
@@ -44,10 +44,11 @@ The Edge Network extension depends on the following extensions:
...
 
-```gradle
-implementation 'com.adobe.marketing.mobile:core:2.+'
-implementation 'com.adobe.marketing.mobile:edge:2.+'
-implementation 'com.adobe.marketing.mobile:edgeidentity:2.+'
+```kotlin
+implementation(platform("com.adobe.marketing.mobile:sdk-bom:3.+"))
+implementation("com.adobe.marketing.mobile:core")
+implementation("com.adobe.marketing.mobile:edge")
+implementation("com.adobe.marketing.mobile:edgeidentity")
```
Comment (emdobrin) [inline Documentation/getting-started.md:51]: can we remove the parenthesis and use a format similar to our documented examples - https://developer.adobe.com/client-sdks/resources/migration/android/migrate-to-2x/#gradle

Comment (kevinlind) [reply Documentation/getting-started.md:51]: As we changed our build files to use Kotlin, I changed them from the Groovy format to the Kotlin format. Android is also asking users to migrate to Kotlin, https://developer.android.com/build/migrate-to-kotlin-dsl#:~:text=Kotlin%20is%20preferred%20over%20the,time%20checking%20and%20IDE%20support.

**Review by cacheung — APPROVED** [review]

File: Documentation/getting-started.md (hunk 2 above)
Comment (emdobrin) [reply Documentation/getting-started.md:51]: do we need to indicate the format used somewhere, and do we need to document both the formats from now on? cc @praveek

**Pull request merged**